shell/browser/native_window_mac.mm

```diff
@@ -326,8 +326,10 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 
   RegisterDeleteDelegateCallback(base::BindOnce(
       [](NativeWindowMac* window) {
-        if (window->window_)
+        if (window->window_) {
+          window->buttons_proxy_.reset(nil);
           window->window_ = nil;
+        }
```

Comment:
Maybe we make these independent, but lgtm
```suggestion
        if (window->window_)
          window->window_ = nil;
        if (window->buttons_proxy_)
          window->buttons_proxy_.reset(nil);
```

Reply:
makes sense!